core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -86,6 +91,23 @@ public Long getDeadlineNanoTime() {
     return deadlineNanoTime;
   }
 
+  /**
+   * Retruns the Codec used for compression and decompression, or {@code null} if none is set.
+   */
+  @Nullable
+  public Codec getCompression() {
```

Comment:
We _need_ compression in the name, otherwise it seems like a marshaller. I would fully expect a `MessageEncoding.Codec` to be "proto" or similar.